server/src/actions/acr.ts

```
@@ -0,0 +1,96 @@
+import { Request, Response } from 'express';
+import { IncomingMessageHeaders } from 'http';
+import Axios from 'axios';
+import { LogHelper } from '../logHelper';
+
+interface ACRDirectRequestPayload {
+    subId: string;
+    endpoint: string;
+    username: string;
+    password: string;
+}
+
+export async function getAcrRepositories(req: Request, res: Response) {
+    const payload = req.body as ACRDirectRequestPayload;
+    
+    if (isRequestPayloadValid(payload, res)) {
+        const responseData = await callAcrAPI(payload);
+        if (responseData) {
+            res.send(responseData);
+        } else {
+            res.status(400).send('Failed to get repositories');
+        }

```

Let's make sure we're not masking errors.  Ideally the nodejs layer should mostly be a passthrough.  See [https://github.com/axios/axios#handling-errors](https://github.com/axios/axios#handling-errors)